platform_plugin_superset/superset_xblock.py

```diff
@@ -0,0 +1,330 @@
+"""XBlock to embed a Superset dashboards in Open edX."""
+from __future__ import annotations
+
+import logging
+from typing import Tuple
+
+import pkg_resources
+from django.conf import settings
+from django.utils import translation
+from web_fragments.fragment import Fragment
+from xblock.core import XBlock
+from xblock.fields import List, Scope, String
+from xblockutils.resources import ResourceLoader
+
+from platform_plugin_superset.extensions.filters import update_context
+from platform_plugin_superset.utils import _
+
+log = logging.getLogger(__name__)
+loader = ResourceLoader(__name__)
+
+
+@XBlock.wants("user")
+@XBlock.needs("i18n")
+class SupersetXBlock(XBlock):
+    """
+    Superset XBlock provides a way to embed dashboards from Superset in a course.
+    """
+
+    display_name = String(
+        display_name=_("Display name"),
+        help=_("Display name"),
+        default="Superset Dashboard",
+        scope=Scope.settings,
+    )
+
+    superset_internal_url = String(
+        display_name=_("Superset URL"),
+        help=_(
+            """Superset internal URL to generate authentication information.
+            Contact with your Open edX operator for more information."""
+        ),
+        default="http://superset:8088/",
+        scope=Scope.settings,
+    )
+
+    superset_url = String(
+        display_name=_("Superset URL"),
+        help=_("Superset URL to embed the dashboard."),
+        default="http://superset.local.overhang.io:8088/",
+        scope=Scope.settings,
+    )
+
+    superset_username = String(
+        display_name=_("Superset Username"),
+        help=_("Superset Username"),
+        default="",
+        scope=Scope.settings,
+    )
+
+    superset_password = String(
+        display_name=_("Superset Password"),
+        help=_("Superset Password"),
+        default="",
+        scope=Scope.settings,
+    )
+
+    dashboard_uuid = String(
+        display_name=_("Dashboard UUID"),
+        help=_(
+            "The ID of the dashboard to embed. Available in the embed dashboard UI."
+        ),
+        default="1d6bf904-f53f-47fd-b1c9-6cd7e284d286",
```

Comment:
why is this the default? can't we use an empty string?

Reply:
Yes, it's for development only

Reply:
is this the default used in development for all environments?

Reply:
I found it https://github.com/opus-42/superset-api-client#usage. It's only needed for development. I'll update the implementation

Reply:
Fixed in #3